d2/src/main/java/com/linkedin/d2/xds/XdsChannelFactory.java

```diff
@@ -61,20 +66,28 @@ public XdsChannelFactory(SslContext sslContext, String xdsServerUri)
    * @param defaultLoadBalancingPolicy If provided, changes the default load balancing policy on the builder to the
    *                                   given policy (see
    *                                   {@link io.grpc.ManagedChannelBuilder#defaultLoadBalancingPolicy(String)}).
-   *                                   Otherwise, by default it uses the "pick_first" policy, with the
-   *                                   "shuffleAddressList" config set to {@code true}. This will make the managed
-   *                                   channel always pick a random xDS server address.
+   * @param loadBalancingPolicyConfig  Can only be provided if {@code defaultLoadBalancingPolicy} is provided. Will be
+   *                                   provided to {@link io.grpc.ManagedChannelBuilder#defaultServiceConfig(Map)}})
+   *                                   after being wrapped in a "loadBalancingConfig" JSON context that corresponds
+   *                                   to the load balancing policy name provided by {@code defaultLoadBalancingPolicy}.
    * @see <a href="https://daniel.haxx.se/blog/2012/01/03/getaddrinfo-with-round-robin-dns-and-happy-eyeballs/"/>
    * Details on IPv6 routing.
    */
   public XdsChannelFactory(
       @Nullable SslContext sslContext,
       String xdsServerUri,
-      @Nullable String defaultLoadBalancingPolicy)
+      @Nullable String defaultLoadBalancingPolicy,
+      @Nullable Map<String, ?> loadBalancingPolicyConfig)
   {
     _sslContext = sslContext;
     _xdsServerUri = xdsServerUri;
+    if (defaultLoadBalancingPolicy == null && loadBalancingPolicyConfig != null)
+    {
+      throw new IllegalArgumentException("Cannot specify loadBalancingPolicyConfig without specifying " +
```

Comment:
Doesn't seem to be a fatal error that needs to throw exception and crash the app. Can it just log a warning instead? the config will just be ignored in this case.

Reply:
Sure